wandb/sdk/artifacts/artifact.py

```diff
@@ -1690,22 +1690,27 @@ def download(
         root = root or self._default_root()
         self._add_download_root(root)
 
-        if wandb.run is None:
-            with wandb.init(  # type: ignore
-                entity=self._source_entity,
-                project=self._source_project,
-                job_type="auto",
-                settings=wandb.Settings(silent="true"),
-            ):
+        if get_core_path() != "":
+            if wandb.run is None:
+                with wandb.init(  # type: ignore
+                    entity=self._source_entity,
+                    project=self._source_project,
+                    job_type="auto",
+                    settings=wandb.Settings(silent="true"),
+                ):
+                    return self._run_artifact_download(
+                        root=root,
+                        allow_missing_references=allow_missing_references,
+                    )
+            else:
                 return self._run_artifact_download(
                     root=root,
                     allow_missing_references=allow_missing_references,
                 )
-        else:
-            return self._run_artifact_download(
-                root=root,
-                allow_missing_references=allow_missing_references,
-            )
+        return self._download(
+            root=root,
+            allow_missing_references=allow_missing_references,
+        )
 
     def _run_artifact_download(
```

Comment:
Nit: this is now only used by `core`, maybe rename to `_download_using_core`. And remove the `if` on line 1722, and else on line 1748.